core/src/main/java/overflowdb/AdjacentNodes.java

```diff
@@ -0,0 +1,29 @@
+package overflowdb;
+
+import overflowdb.util.PackedIntArray;
+
+public class AdjacentNodes {
+
+  /**
+   * holds refs to all adjacent nodes (a.k.a. dummy edges) and the edge properties
+   */
+  public final Object[] nodesWithEdgeProperties;
+
+  /* store the start offset and length into the above `adjacentNodesWithEdgeProperties` array in an interleaved manner,
+   * i.e. each adjacent edge type has two entries in this array. */
+  public final PackedIntArray edgeOffsets;
```

Comment:
I think we should inline the PackedIntArray here: That's otherwise one extra indirection that is unlikely to be in cache, and main memory is far far away.

Reply:
as discussed, via follow-up pr